app/models/cdr/table.rb

```diff
@@ -53,15 +55,40 @@ def archive
 
   def self.add_partition
     transaction do
-      self.execute_sp("SELECT * FROM sys.cdr_createtable(2)")  ##try to create table for previous month
-      self.execute_sp("SELECT * FROM sys.cdr_createtable(1)")  ##try to create table for current month
-      self.execute_sp("SELECT * FROM sys.cdr_createtable(0)") ##for next month
+      today = Date.today
+      # TODO: find out what design pattern suits here
+      Yeti::PgTablePartition.new(connection: connection, date_from: today.prev_month).create_partition
+      Yeti::PgTablePartition.new(connection: connection, date_from: today).create_partition
+      Yeti::PgTablePartition.new(connection: connection, date_from: today.next_month).create_partition
     end
   end
 
+  def self.destroy_partitions
```

Comment:
This method should raise exception if application running in production mode. It is not save to allow drop all data in production